File: api/src/models/song.ts
import * as t from 'io-ts';

export const SongModelV = t.type({
  id: t.number,
  artists: t.array(t.string),
  title: t.string,
  albumArt: t.union([t.string, t.null]),
  spotifyId: t.union([t.string, t.null]),
  appleMusicId: t.union([t.string, t.null]),
  appleMusicUrl: t.union([t.string, t.null]),
  isrcId: t.union([t.string, t.null]),
  album: t.union([t.string, t.null]),
});

export type SongModel = t.TypeOf<typeof SongModelV>;

File: rhiannon/src/main/kotlin/club/jambuds/model/Song.kt
package club.jambuds.model

import com.fasterxml.jackson.annotation.JsonIgnore
import java.time.Instant

data class Song(
    val id: Int,
    @JsonIgnore val createdAt: Instant,
    val title: String,
    val artists: List<String>,
    val album: String?,
    val albumArt: String?,
    val spotifyId: String?,
    val isrcId: String?,
    val appleMusicId: String?,
    val appleMusicUrl: String?,
    val bandcampId: String?,
    val bandcampUrl: String?
)
